cadence/contracts/bridge/FlowEVMBridgeUtils.cdc

```diff
@@ -727,6 +781,198 @@ contract FlowEVMBridgeUtils {
+
+        // Ensure bridge to recipient was succcessful
+        let toPostBalance = FlowEVMBridgeUtils.balanceOf(owner: to, evmContractAddress: erc20Address)
+        let escrowPostBalance = FlowEVMBridgeUtils.balanceOf(
+            owner: bridgeCOAAddress,
+            evmContractAddress: erc20Address
+        )
+        assert(
+            toPostBalance == toPreBalance + amount,
+            message: "Transfer to bridge COA failed - cannot bridge FT without bridge escrow"
+        )
+        assert(
+            escrowPostBalance == escrowPreBalance - amount,
+            message: "Transfer to bridge COA failed - cannot bridge FT without bridge escrow"
+        )
+    }
+
+    /// Executes the provided method, assumed to be a protected transfer call, and confirms that the transfer was
+    /// successful by validating that the named owner's balance was decremented by the requested amount and the bridge
+    /// escrow balance was incremented by the same amount.
+    ///
+    access(account)
+    fun mustExecuteProtectedERC20TransferCall(
+        owner: EVM.EVMAddress,
+        amount: UInt256,
+        erc20Address: EVM.EVMAddress,
+        protectedTransferCall: fun (): EVM.Result
+    ) {
+        // Ensure the caller is has sufficient balance to bridge the requested amount
+        let hasSufficientBalance = FlowEVMBridgeUtils.hasSufficientBalance(
+            amount: amount,
+            owner: owner,
+            evmContractAddress: erc20Address
+        )
+        assert(hasSufficientBalance, message: "Caller does not have sufficient balance to bridge requested tokens")
+
+        // Get the owner and escrow balances before transfer
+        let ownerPreBalance = FlowEVMBridgeUtils.balanceOf(owner: owner, evmContractAddress: erc20Address)
+        let bridgePreBalance = FlowEVMBridgeUtils.balanceOf(
+                owner: FlowEVMBridgeUtils.getBridgeCOAEVMAddress(),
+                evmContractAddress: erc20Address
+            )
+
+        // Call the protected transfer function which should execute a transfer call from the owner to escrow
+        let transferResult = protectedTransferCall()
+        assert(transferResult.status == EVM.Status.successful, message: "Transfer via callback failed")
+
+        // Get the resulting balances after transfer
+        let ownerPostBalance = FlowEVMBridgeUtils.balanceOf(owner: owner, evmContractAddress: erc20Address)
+        let bridgePostBalance = FlowEVMBridgeUtils.balanceOf(
+                owner: FlowEVMBridgeUtils.getBridgeCOAEVMAddress(),
+                evmContractAddress: erc20Address
+            )
+        
+        // Confirm the transfer of the expected was successful in both sending owner and recipient escrow
+        assert(ownerPostBalance == ownerPreBalance - amount, message: "Transfer to owner failed")
+        assert(bridgePostBalance == bridgePreBalance + amount, message: "Transfer to bridge escrow failed")
```

Comment:
Same comments about these error messages, I feel like they should be more descriptive